src/tests/ftest/util/dfuse_utils.py

```diff
@@ -0,0 +1,224 @@
+
+        Args:
+            hosts: list of hosts where dfuse directory needs to be created
+
+        Raises:
+            CommandFailure: In case of error creating directory
+        """
+        for host in hosts:
+            if general_utils.check_file_exists(
+                    [host], self.mount_dir.value, directory=True)[0] is False:
+                ret_code = general_utils.pcmd(
+                    [host], "mkdir " + self.mount_dir.value, timeout=30)
+                for key in ret_code:
+                    if key is not 0:
+                        raise CommandFailure(
+                            "DfuseFailure: Error creating directory: "
+                            "{}".format(self.mount_dir.value))
+
+    def remove_dfuse_dir(self, hosts):
+        """Remove dfuse directory
+
+        Args:
+            hosts: list of hosts from where dfuse directory needs to be deleted
+
+        Raises:
+            CommandFailure: In case of error deleting directory
+        """
+        if general_utils.check_file_exists(
+                hosts, self.mount_dir.value)[0] is True:
+            ret_code = general_utils.pcmd(
+                hosts, "rm -rf " + self.mount_dir.value, timeout=30)
+            for key in ret_code:
+                if key is not 0:
+                    raise CommandFailure("DfuseFailure: Error deleting: "
+                                         "{}".format(self.mount_dir.value))
+
+    def run_dfuse(self, hosts, attach_info, basepath):
+        """ Run the dfuse command.
+
+        Args:
+            hosts: list of hosts where dfuse needs to be started
+            attach_info (str): CART attach info path
+            basepath: path for daos install dir
+        Raises:
+            CommandFailure: In case dfuse run command fails
+        """
+
+        # create dfuse dir if does not exist
+        self.create_dfuse_dir(hosts)
+        # obtain env export string
+        env = self.get_default_env(attach_info, basepath)
+
+        # run dfuse command
+        ret_code = general_utils.pcmd(hosts, env + self.__str__(), timeout=30)
+
+        # check for any failures
+        for key in ret_code:
+            if key is not 0:
+                raise CommandFailure("DfuseFailure: Error starting dfuse. "
+                                     "RC:{}".format(ret_code))
+
+    def stop_dfuse(self, hosts):
+        """Stop dfuse
+
+        Args:
+            hosts: list of hosts where dfuse needs to be stopped
+
+        Raises:
+            CommandFailure: In case dfuse stop fails
+        """
+
+        for host in hosts:
+            if 0 in general_utils.pcmd([host], "which fusermount"):
+                fuse_cmd = "fusermount -u {}".format(self.mount_dir.value)
+            elif 0 in general_utils.pcmd([host], "which fusermount3"):
+                fuse_cmd = "fusermount3 -u {}".format(self.mount_dir.value)
+            else:
+                raise CommandFailure("No Fuse on {}".format([host]))
```

Comment:
It feels like this should be done in a single `pcmd(self.hosts, cmd, timeout=60)` call, where the `cmd` does something like  `"if [ -x $(command -v fusermount) ]; then fusermount -u {0}; else; fusermount3 -u {0}; fi".format(self.mount_point)`